base_changeset/models/record_changeset_change.py

```diff
@@ -371,9 +372,10 @@ def get_fields_changeset_changes(self, model, res_id):
         states = self.get_pending_changes_states()
         domain = [
             ("changeset_id.model", "=", model),
-            ("changeset_id.res_id", "=", res_id),
             ("state", "in", states),
         ]
+        if res_id:
+            domain.append(("changeset_id.res_id", "=", res_id))
```

Comment:
I think you have to add a domain "impossible" when you don't specify `res_id`, otherwise you will get all records (something wrong).  Could this be the cause of https://github.com/OCA/server-tools/pull/2184#issuecomment-963081075 ?